internal/parse/parse.go

```diff
@@ -507,18 +513,23 @@ func getMounts(ctx *types.SystemContext, store storage.Store, mounts []string, c
 	//                  to allow a more robust parsing of the mount format and to give
 	//                  precise errors regarding supported format versus supported options.
 	for _, mount := range mounts {
-		arr := strings.SplitN(mount, ",", 2)
+		arr := strings.Split(mount, ",")
 		if len(arr) < 2 {
 			return nil, mountedImages, lockedTargets, fmt.Errorf("%q: %w", mount, errInvalidSyntax)
 		}
-		kv := strings.Split(arr[0], "=")
+		typeToken := arr[0]
+		for _, field := range arr {
+			if strings.HasPrefix(field, "type=") {
+				typeToken = field
+			}
+		}
+		kv := strings.Split(typeToken, "=")
 		// TODO: type is not explicitly required in Docker.
-		// If not specified, it defaults to "volume".
+		// If not specified, it defaults to "bind".
 		if len(kv) != 2 || kv[0] != "type" {
 			return nil, mountedImages, lockedTargets, fmt.Errorf("%q: %w", mount, errInvalidSyntax)
```

Comment:
It looks like this might complain if the mount specification doesn't include a "type" at all.  Please ensure there's a test that uses neither "from" nor "type".